Configure.py

```diff
@@ -67,51 +109,111 @@ def main(argv):
     # System configuration
     port = None
     # User preferences
+    local = None
```

Comment:
Isn't it possible to set default values in the 'args' parameters, possibly even centrally, so no program should have to worry about initializing these variables but can instead always rely on 'args.xyz' (which IIRC will be 'None' anyway if no default value was supplied)?

Reply:
The default values are in the Config object. These are used if there hasn't been a value configured and saved using Configure.

Maybe I should create an 'action' object used in the argument overrides provided by the Config object. That way any common options specified would directly update the Config object. Then the application would read any needed Config values from the Config object (like config.text_speed, config.port, etc.) rather than reading them from the args. The only args the application would access/process are the options specific to the application.

Reply:
I will work on creating that and see how it works.

Reply:
@pwdirks - Per the conversation in the issue comments, I will leave this as is for now and we can discuss approaches for later.

Currently, the values are set by:  
1. Defaults coded in the Config object
2. Values configured by the user using Configure.py (or manually editing the INI file)

These values are then used to initialize the values for the command line option processing.

Does that explaination resolve this, or is there something else that you are suggesting?

Reply:
[ I think another emailed response was lost.  Repeating here: ]

The Config object has default values for all of the parameters. It has a default of 18 for both min_char_speed and text_speed. So you will get those values from config.char_speed_min and config.text_speed if the user hasn't run Configure. You will get the values that are in the INI file if they exist.

The option-overrides in Config set a default from the value in the Config object, they should have a value based on the default or the value saved in the INI file (set by running Configure).


That sounds sensible.  So you can always pick out ‘arg.xyz’ for the specified version, either as default (absent a config), as default for the argument (from the config) or as specified by the user (if overridden).
Maybe we should create an action class that the option-overrides use to set the specified values on the Config object directly, such that all of the 'standard' options (that are set as parents by the application) update the Config object and no action is required by the application for those options. The application would only processes the options it has added. Then the application pulls all of the config values from the Config object (rather than from the args).


If I follow you correctly you’re proposing to side-effect the config object when I override an argument, just so that I can always reach for the ‘config.xyz’ value to get the desired value for a given run?  I think that would be very counter-intuitive - I don’t expect to change the default speed just because I, say, specify 25 WPM on a run.  The config should be what I configured, but I think that DOES make the perfect default for an argument override.

Hope that makes sense?